microdata_tools/packaging/__init__.py

```
@@ -122,15 +118,14 @@ def unpackage_dataset(
         untar_encrypted_dataset(packaged_file_path, dataset_name, dataset_dir)
         decrypt(rsa_keys_dir, dataset_dir, output_dir)
         _validate_csv_consistency(dataset_name, dataset_dir, output_dir)
-        if archive_dir is not None:
-            _archive(
-                dataset_name, dataset_dir.parent, archive_dir, "unpackaged"
-            )
+
+        if Path(dataset_dir).exists():
+            shutil.rmtree(dataset_dir)
+
         logger.info(f"Unpackaged {packaged_file_path}")
     except Exception as exc:
-        if archive_dir is not None:
-            _archive(dataset_name, dataset_dir.parent, archive_dir, "failed")
         logger.exception(f"Failed to unpackage {dataset_name}", exc_info=exc)
+        raise exc
```

Can you raise a named exception here like:
```py
raise UnpackagingError("Failed to unpackage dataset") from exc
```
?